pkg/review/commit_review_status.go

```diff
@@ -91,6 +106,71 @@ type PullRequest struct {
 	ReviewDecision githubv4.String
 }
 
+// CommitApprovalPipelineConfig holds the configuration data for the
+// commit approval beam pipeline.
+type CommitApprovalPipelineConfig struct {
+	// The token to use for authenticating with GitHub.
+	GitHubAccessToken string
+	// The GCP project that holds the BigQuery tables
+	BigQueryProject string
+	// The name of the BigQuery dataset that contains the tables
+	BigQueryDataset string
+	// The name of the BigQuery table that holds push event data. This is the
+	// table that is used to source the commits that need to be processed.
+	BigQueryPushEventsTable string
+	// The name of the BigQuery table that holds the commit review/approval status.
+	// This is the table that stores the final output of pipeline.
+	BigQueryCommitReviewStatusTable string
+	// The name of the BigQuery table that holds GitHub issue data. This table
+	// is used to determine if a commit was pushed using 'break glass' permissions.
+	BigQueryIssuesTable string
+}
+
+// CommitApprovalDoFn is an object that implements beams "DoFn" interface to
+// provide the processing logic for converting a Commit to CommitReviewStatus.
+type CommitApprovalDoFn struct {
+	Config       CommitApprovalPipelineConfig
+	GithubClient *githubv4.Client
+}
+
+// ProcessElement is a DoFn implementation that take a Commit, determines
+// if the commit was properly approved, and outputs the resulting
+// CommitReviewStatus using the provided emit function.
+// A commit is considered properly reviewed as long as there is an associated
+// PR for the commit targeting the repository's main branch with reviewDecision
+// of 'APPROVED'.
+func (fn *CommitApprovalDoFn) ProcessElement(ctx context.Context, commit Commit, emit func(CommitReviewStatus)) {
+	logger := logging.FromContext(ctx)
+	logger.Infow("processing commit", "commit", commit)
+	requests, err := GetPullRequestsTargetingDefaultBranch(ctx, fn.GithubClient, commit.Organization, commit.Repository, commit.SHA)
+	if err != nil {
+		logger.Errorf("failed to get pull requests for commit: %v", err)
+		return // this commit could not be processed
```

Comment:
It's a little concerning that this error isn't handled in any way besides logging

Reply:
There are essentially two different kind of errors that could happen:
1. **Transient Errors**: We aren't able to get the pull requests for a commit because of some temporary issue with GitHub (e.g. GitHub servers are down). In these cases the commit will simply be retried the next time the pipeline is run.
2. **Perment Errors**: There is something wrong with the commit data itself that makes GitHub return an error. For example, the repository the commit came from may have been deleted so we no longer can get pull request information from GitHub about it.

For the **Transient Errors**, the commit will be retried during the next pipleine execution so there is not need to do anything else aside from logging the error. For **Perment Errors** it may be useful to do something aside from logging, but it is hard to say exactly what should be done without seeing what kinds of errors like this occur and how frequently. My thought was to log the error for now and then consider more sophisticated error handling if we need it.

Reply:
Could you add this as a code comment?

Reply:
done.